### ISSUE TITLE
The train not save checkpoint

### ISSUE THREAD

**Issue by zplovekq** [pytorch/fairseq#2130]
When i train with save-interval-updates 500 and  --disable-validation. 
Then after the update 500, the checkpoint not save.
But when i only with save-interval-updates 1000, some times the train epoch not finish yet,then the valid start .After the valid finish ,  the next epoch start.

**Comment by myleott** [comment]
Can you share the full command you ran? I tried a dummy model and it worked:
```bash
python train.py \
  --task dummy_masked_lm --dict-size 49995 \
  --max-sentences 2 --arch roberta_base \
  --log-interval 1 --log-format simple \
  --save-interval-updates 10 --disable-validation
```

**Comment by zplovekq** [comment]
Thanks for your reply!
python -m fairseq_cli.train   --restore-file checkpoint30.pt  --save-dir .checkpoints-scst-3 --task captioning   --arch default-captioning-arch    --optimizer adam   --adam-betas "(0.9,0.999)"   --reset-optimizer   --lr 5e-6   --weight-decay 0.0001   --criterion self_critical_sequence_training   --dropout 0.3   --max-epoch 20 --max-sentences 1 --max-source-positions 100   --max-target-positions 50   --encoder-embed-dim 512   --tokenizer moses   --bpe subword_nmt   --bpe-codes output/codes.txt   --ddp-backend=no_c10d   --scst-beam 5   --scst-penalty 1.0   --scst-validation-set-size 0 --num-workers 0 --save-interval-updates 1000 
I use the command line before.And some times the epoch 31 has not finish yet, then the valid start,then go to the epoch 32.
And some times , i find the GPU memory are not the same, like this:
![image](https://user-images.githubusercontent.com/39880124/82006284-0a4a6880-969a-11ea-9bf7-8b507c0151bc.png)
then the train will stop. Some times log will output OOM Error and some times the log will not just output.
Thanks for any guide!

**Comment by zplovekq** [comment]
I occur this issue again.Like this:
![image](https://user-images.githubusercontent.com/39880124/82010802-b34a9080-96a5-11ea-9ae9-2be89679fed6.png)
As the log , the train epoch 33 has not finish yet , the epoch 34 start. The command line as i write before.

**Comment by myleott** [comment]
Hmm, so this seems like something could be wrong with your Task/Dataset. Since you're using a custom Dataset, make sure that the `__len__` attribute is correct, since that's what we use to populate the progress bar. Are you able to share the Task+Dataset code somewhere?
